elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/metadata/SQLReferenceTable.java

```diff
@@ -98,7 +97,13 @@ public String getResolvedReference(Queryable queryable, String fieldName) {
      * @return resolved ON clause expression
      */
     public Set<String> getResolvedJoinExpressions(Queryable queryable, String fieldName) {
-        return resolvedJoinExpressions.get(queryable).get(fieldName);
+        Set<String> joinExpressions = resolvedJoinExpressions.get(queryable).get(fieldName);
+
+        if (joinExpressions == null) {
+            return new HashSet<>();
+        }
+
+        return joinExpressions;
```

Comment:
may be change to `return resolvedJoinExpressions.get(queryable).getOrDefault(fieldName, new HashSet<>());` as below.